tools/sgcollect_info

```diff
@@ -385,9 +395,22 @@ def make_collect_logs_tasks(
         os_home_dirs.append(guessed_log_path)
 
     # Keep a dictionary of log file paths we've added, to avoid adding duplicates
-    sg_log_file_paths = {}
-
-    sg_tasks = []
+    sg_log_file_paths: set[pathlib.Path] = set()
+
+    sg_tasks: List[PythonTask] = []
+
+    def add_file_collection_task(filename: pathlib.Path):
+        canonical_filename = pathlib.Path(filename)
+        canonical_filename.resolve()
+        if canonical_filename in sg_log_file_paths:
+            return
+        sg_log_file_paths.add(canonical_filename)
+        if not should_redact or not redactable_file(filename):
+            task = add_file_task(sourcefile_path=filename)
+            task.no_header = True
+            sg_tasks.append(task)
+        else:
+            sg_tasks.append(add_gzip_file_task(sourcefile_path=filename, salt=salt))
```

Comment:
Not all of these files are actually gzipped. For example, this glob: `"{0}*{1}".format(name, ext))` grabs `sg_info.log`

It used to be run through `add_file_task` instead of `add_gzip_file_task`